### DISCUSSION TITLE
How do I specify crop images in order?

### DISCUSSION THREAD

**Discussion by cvlearn913** [Project-MONAI/MONAI#3707] [Q&A] [answered]
Monai has the following transforms, I have a (512,512,512) size of data that I want to crop to 4 patches in order. The ROI locations is that [0-127,0-127,0-127],[128-255,128-255,128-255],[256-384,256-384,256-384],[385-511,385-511,385-511]. What should I do?

```python
SpatialCrop
CenterSpatialCrop
RandSpatialCrop
RandSpatialCropSamples
CropForeground
RandWeightedCrop
RandCropByPosNegLabel
RandCropByLabelClasses
ResizeWithPadOrCrop
BoundingRect
RandScaleCrop
CenterScaleCrop

```

**Comment by Nic-Ma** [comment, marked as answer]
Hi @cvlearn913 ,

Thanks for your interest here.
I think you can (1) add channel dim, (2) copy the original image 3 times, (3) apply `SpatialCrop` transform to them with expected ROI start / end.

Thanks.

**Comment by cvlearn913** [comment]
Thank you sincerely for your kind regards！ @Nic-Ma But it seems that dictionary ```spatialcropd```  [functions](https://docs.monai.io/en/stable/transforms.html?highlight=spatialcropd#spatialcropd) do not support multiple clipping. I want to crop and get four samples, but only get one sample. 
```python
transforms = Compose(
    [     LoadImaged(keys=["image", "label"]),
          EnsureChannelFirstd(keys=["image", "label"]),
          Resized(keys=["image","label"],spatial_size=[512,512,512]),
          SpatialCropd(keys=["image","label"],roi_size=[128,128,128],roi_slices=[slice(0,128,1),slice(0,128,1),slice(0,128,1)]),
          # The data has already changed to 128*128*128
          EnsureTyped(keys=["image", "label"]),
            
    ]
)

**Reply by Nic-Ma** [reply]
Hi @cvlearn913 ,

I think you missed the (2) step in my previous comment, you can try something like:
```py
LoadImaged(keys="img"),
EnsureChannelFirstd(keys="img"),
CopyItemsd(keys="img", times=3),
SpatialCropd(keys="img", roi_start=..., roi_end=...),
SpatialCropd(keys="img_0", roi_start=..., roi_end=...),
SpatialCropd(keys="img_1", roi_start=..., roi_end=...),
SpatialCropd(keys="img_2", roi_start=..., roi_end=...),
```
Thanks.

**Reply by cvlearn913** [reply]
Thanks it works ! 
![image](https://user-images.githubusercontent.com/59678550/150947001-297c7b59-0a0d-4010-8895-a6d6385383ce.png)

**Reply by Nic-Ma** [reply]
Seems your `image_0` and `image_1`, `image_2` are the same data content, is it expected?

Thanks.

**Reply by cvlearn913** [reply]
Yeah, I set the roi_start ，roi_end all the same for testing

**Reply by Nic-Ma** [reply]
Cool!

**Reply by AnupKumarGupta** [reply]
Hello. Thank you for your solution. I too, have a similar query, but instead of just 3 patches, I require 64 patches. Would there be any other way to do so? Or should I copy the original image 64 times? Thanks in advance.

**Reply by KumoLiu** [reply]
Hi @AnupKumarGupta, I think you could try to use `GridPatch` and `SplitDim`.
https://github.com/Project-MONAI/MONAI/blob/6803061769a2dcec5839887967b0be0f2b34ac7a/monai/transforms/spatial/array.py#L3176
https://github.com/Project-MONAI/MONAI/blob/6803061769a2dcec5839887967b0be0f2b34ac7a/monai/transforms/utility/array.py#L334

A simple example here:
```
from monai.transforms import GridPatch, SplitDim, Compose

trans = Compose([
    GridPatch(num_patches=64),
    SplitDim(dim=0)
])
out = trans(data)
```
Hope it can help you, thanks!

**Reply by AnupKumarGupta** [reply]
@KumoLiu Thanks a lot.
